cmd/gitsign-credential-cache/main.go

```diff
@@ -47,31 +49,53 @@ func main() {
 		os.Exit(0)
 	}
 
-	user, err := os.UserCacheDir()
-	if err != nil {
-		log.Fatalf("error getting user cache directory: %v", err)
-	}
+	var connChan = make(chan net.Conn)
+	if *systemdFlag {
+		listeners, err := activation.Listeners()
+		if err != nil {
+			log.Fatalf("error getting systemd listeners: %v", err)
+		}
+		for _, l := range listeners {
+			go connToChan(l, connChan)
+		}
+	} else {
+		user, err := os.UserCacheDir()
+		if err != nil {
+			log.Fatalf("error getting user cache directory: %v", err)
+		}
 
-	dir := filepath.Join(user, "sigstore", "gitsign")
-	if err := os.MkdirAll(dir, 0700); err != nil {
-		log.Fatalf("error creating %s: %v", dir, err)
-	}
+		dir := filepath.Join(user, "sigstore", "gitsign")
+		if err := os.MkdirAll(dir, 0700); err != nil {
+			log.Fatalf("error creating %s: %v", dir, err)
+		}
 
-	path := filepath.Join(dir, "cache.sock")
-	if _, err := os.Stat(path); err == nil {
-		os.Remove(path)
-	}
-	fmt.Println(path)
+		path := filepath.Join(dir, "cache.sock")
+		if _, err := os.Stat(path); err == nil {
+			os.Remove(path)
+		}
+		fmt.Println(path)
```

Comment:
This is kinda pointless: sockets are configured in systemd unit, program gets only file descriptors. And if you want to know what's current socket configuration, then you can just run `systemctl --user status gitsign-credential-cache.socket` and get something like this
```
gitsign-credential-cache.socket - GitSign credential cache socket
     Loaded: loaded (/home/iavael/.config/systemd/user/gitsign-credential-cache.socket; enabled; preset: disabled)
     Active: active (running) since Tue 2023-01-17 17:50:53 +04; 1 day 9h ago
      Until: Tue 2023-01-17 17:50:53 +04; 1 day 9h ago
   Triggers: ● gitsign-credential-cache.service
     Listen: /home/iavael/.cache/sigstore/gitsign/cache.sock (Stream)
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/gitsign-credential-cache.socket

янв 17 17:50:53 workstation systemd[5417]: Listening on gitsign-credential-cache.socket - GitSign credential cache socket.
```
In "Listen" field there will be all listening sockets

But I added printing them just in case.